packages/alert-banner/src/AlertBanner.ts

```diff
@@ -0,0 +1,146 @@
+    public set variant(variant: AlertBannerVariants) {
+        if (variant === this.variant) {
+            return;
+        }
+        const oldValue = this.variant;
+        if (this._isValidVariant(variant)) {
+            this.setAttribute('variant', variant);
+            this._variant = variant;
+        } else {
+            this.removeAttribute('variant');
+            this._variant = '';
+        }
+        this.requestUpdate('variant', oldValue);
+    }
+
+    public get variant(): AlertBannerVariants {
+        return this._variant;
+    }
+
+    private _variant: AlertBannerVariants = '';
+
+    protected _isValidVariant(variant: string): boolean {
+        return VALID_VARIANTS.includes(variant);
+    }
+
+    protected renderIcon(variant: string): TemplateResult {
+        switch (variant) {
+            case 'info':
+                return html`
+                    <sp-icon-info
+                        label="Information"
+                        class="type"
+                    ></sp-icon-info>
+                `;
+            case 'negative':
+                return html`
+                    <sp-icon-alert label="Error" class="type"></sp-icon-alert>
+                `;
+            default:
+                return html``;
+        }
+    }
+
+    private _shouldClose(): void {
+        const applyDefault = this.dispatchEvent(
+            new CustomEvent('close', {
+                composed: true,
+                bubbles: true,
+                cancelable: true,
+            })
+        );
+        if (applyDefault) {
+            this.close();
+        }
+    }
+
+    public close(): void {
+        this.open = false;
+    }
+
+    private _handleKeydown(event: KeyboardEvent): void {
+        if (event.code === 'Escape' && this.dismissible) {
+            this._shouldClose();
+        }
+    }
+
+    protected override render(): TemplateResult {
+        return html`
+            <div class="body" role="alert">
+                <div class="content">
+                    ${this.renderIcon(this.variant)}
+                    <div class="text"><slot></slot></div>
+                </div>
+                <slot name="action"></slot>
+            </div>
+            <div class="end">
+                ${this.dismissible
+                    ? html`
+                          <sp-close-button
+                              @click=${this._shouldClose}
+                              label="Close"
+                              static="white"
+                          ></sp-close-button>
+                      `
+                    : html``}
+            </div>
+        `;
+    }
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.addEventListener('keydown', this._handleKeydown.bind(this));
+    }
+
+    public override disconnectedCallback(): void {
+        super.disconnectedCallback();
+        this.removeEventListener('keydown', this._handleKeydown.bind(this));
```

Comment:
This is a memory leak.

Reply:
You're right, thanks for pointing this out 💯